doc/ocpf-match-action.md

```diff
@@ -0,0 +1,385 @@
++                            +--rw log?                  boolean
++                            +--rw mpls-traffic-class?   oc-mplst:mpls-tc
+                             +--rw network-instance?
+                             +--rw next-hop?
+                             +--rw next-hop-group?
++                            +--rw next-rule?            boolean  # see chaining
+                             +--rw path-selection-group?
+                             +--rw post-decap-network-instance?
++                            +--rw sample?               boolean # packet should be sampled
++                            +--rw ip-ttl?               uint8   # see openconfig/public#1313
+```
+
+## 3.2. User defined counters tree
+
+In order to support the `action/config/count` leaf, a new top level
+user-counters subtree is introduced.
+
+```diff
++    +--rw user-counters
++    |  +--rw user-counter* [name]
++    |  |  +--rw config
++    |        +--rw name?           string
++    |  |  +--rw state
++    |  |     +--rw name?           string
++    |  |     +--rw counters
++    |  |        +--rw count?       oc-yang:counter64
+```
+
+# 4. Multiple Rules and Actions
+
+OC does not explicitly state what logic is used when an OCPF policy has multiple rules, multiple criteria within a rule and if there are multiple actions on a rule.  OC is not clear on what should happen after an OCPF rule is matched or not matched.  The following text is proposed to be added to the OCPF model to describe how to handle multiple rules and actions.
+
+Multiple match criteria in a single rule is treated logically as “AND”.
+
+Multiple rules in a policy should be treated logically as “OR”.
+
+Each rule must have one or more actions.
+
+When a packet matches a rule, all actions associated with a rule are performed.
+
+When a rule is matched, processing stops, unless the `next-rule` leaf is populated.
+
+# 5. Order of operations and mapping match/action rules into device implementations
+
+## 5.1. OC defined order of operations
+
+Today OC defines the following order of operations between ACL, QOS and OCPF.  “Routing” is included as the step expected to follow policy forwarding.  Note that In today’s OC model, the sequence of ACL and QOS is ambiguous; OCPF is defined to happen after ACL and QOS.
+
+![OCPF Summary](img/ocpf-match-action-old-order.png)
+
+In practice, implementations using ASICs for their packet processing compile their “match action rule” (ACL, QoS classifier,  policy forwarding, PBR, firewall, traffic policy, etc…) configurations into banks of TCAM memories.   The rules are compiled as bitfield masks which extract key data from packets.  The keys are looked up in TCAM memory to map to an action.    The TCAM memories are often rather limited in scale and the implementation details of the hardware and programming vary widely between ASICs and NOSs.
+
+Abstractly the programming of these rules can be organized as follows:
```

Comment:
I would suggest to make it clear that this is an example mapping for a theoretical chip. It might be worth demonstrating that this model is independent of chip architecture.